frontend/src/contexts/ZoomUIContext.tsx

```diff
@@ -0,0 +1,47 @@
+"use client";
+import React, { createContext, useContext, useState, useRef, useEffect } from "react";
+import { ZoomOptions } from "@/enums/ZoomOptions";
+
+interface ZoomContextProps {
```

Comment:
I thought we wanted to name the interfaces of contexts the same way as the contexts themselves. If so, it should be `ZoomContext`.
Or we could move this interface into the `interfaces` directory and import it as `IZoomContext`.
If, however, we want to name context interfaces with the `*Props` suffix, then we should be consistent and use this convention for **every** context.